### ISSUE TITLE
False positive for Empty Enum Arguments

### ISSUE THREAD

**Issue by DaniSpringer** [realm/SwiftLint#3908]
`if !CLLocationManager.locationServicesEnabled() {`

**Comment by jpsim** [comment]
can you fill out all the fields you deleted when opening this issue? https://github.com/realm/SwiftLint/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by waydabber** [comment]
This popped up for me lately as well for every occurrence of something like this:

`if !foo.bar() {`

**Comment by waydabber** [comment]
Updating to the latest version fixed that issue for me.

**Comment by jpsim** [comment]
I also believe this is fixed. @DaniSpringer please open a new issue taking care to provide the information requested in the template if you're still seeing this after updating to the latest release.

**Comment by codelobber** [comment]
the same issue.


---
name: Bug report
about: False positive for Empty Enum Arguments on negative boolean functions
---

### New Issue Checklist

- [X ] Updated SwiftLint to the latest version 0.47
- [ X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues) 

### Describe the bug

swiftlint trigger rule `empty_enum_arguments` on if statment when it contain negative boolean function return
bug appear only on systems with Xcode 13.3. 13.2.1 work fine.


##### Complete output when running SwiftLint, including the stack trace and command used

```bash
XXX.swift:137:47: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
XXX.swift:98:20: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
XXX.swift:50:20: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
:0:0: error: Warning Threshold Violation: Number of warnings exceeded threshold of 1. (warning_threshold)
```

### Environment

* SwiftLint version: 0.47.0
* Installation method used: CocoaPods
* Configuration file:

```yml
warning_threshold: 1

opt_in_rules:
  - attributes
#  - closure_end_indentation
  - closure_parameter_position
  - closure_spacing
  - collection_alignment
  - colon
  - convenience_type
  - discouraged_object_literal
  - empty_collection_literal
  - empty_count
  - empty_string
  - enum_case_associated_values_count
  - fatal_error_message
  - first_where
  - force_unwrapping
  - identical_operands
  - implicitly_unwrapped_optional #https://google.github.io/swift/#implicitly-unwrapped-optionals
  - joined_default_parameter
  - last_where
  - legacy_random
  - literal_expression_end_indentation
#  - multiline_arguments
#  - multiline_arguments_brackets
  - multiline_function_chains
  - multiline_literal_brackets
  - multiline_parameters
  - multiline_parameters_brackets
  - operator_usage_whitespace
  - optional_enum_case_matching
  - overridden_super_call
  - pattern_matching_keywords
  - prefer_self_type_over_type_of_self
  - reduce_into
  - redundant_nil_coalescing
  - redundant_type_annotation
  - toggle_bool
#  - trailing_closure
  - unneeded_parentheses_in_closure_argument
  - unused_declaration
  - unused_import
  - vertical_whitespace_closing_braces
  - vertical_whitespace_opening_braces
  - yoda_condition

disabled_rules:
  - private_over_fileprivate
  - weak_delegate
  - trailing_comma
  - trailing_whitespace
  - type_name
  - large_tuple
  - block_based_kvo
  - function_parameter_count
  - function_body_length
  - no_fallthrough_only
  - orphaned_doc_comment
  - todo
  - nesting
  - multiple_closures_with_trailing_closure
  - cyclomatic_complexity
  - inclusive_language

identifier_name:
  min_length: 2
  excluded:
    - i
    - j
    - id
    - vc
    - x
    - y


empty_count:
  severity: warning

force_cast: warning
force_try: warning

line_length:
  warning: 110 #https://google.github.io/swift/#general-formatting
  ignores_urls: true
  ignores_function_declarations: true
  ignores_comments: true

file_length: 400
type_body_length: 450

excluded:
  - Pods/
  - Frameworks/
  - XXX/*/.build/
  - XXX/*/Tests

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. - NO
* Which Xcode version are you using (check `xcodebuild -version`)? - Xcode 13.3 


```swift
if !date.isInCurrentYear↓() {
     title += " \(date.year)"
}
```

**Comment by jpsim** [comment]
Please open a new issue
